playground/src/hooks/useKubrickAPI.ts

```
@@ -202,15 +222,34 @@ export const fetchTasks = async (
 };
 
 // React Query hook for tasks
-export const useGetTasks = (
-  page = 0,
+export const useGetAndPrefetchTasks = (
+  page: number,
   limit: number,
   isAccordionOpen: boolean
-) =>
-  useQuery<TasksResponse, Error>({
+) => {
+  const queryClient = useQueryClient();
+
+  const query = useQuery<TasksResponse, Error>({
     queryKey: ["data", page, limit],
     queryFn: () => fetchTasks(page, limit),
-    placeholderData: (prev) => prev, // Keeps old data during loading,
+    placeholderData: (prev) => prev,
     refetchInterval: 5000,
     enabled: isAccordionOpen,
   });
+
+  useEffect(() => {
+    if (!isAccordionOpen) return;
+
+    queryClient.prefetchQuery({
+      queryKey: ["data", page, limit],
+      queryFn: () => fetchTasks(page, limit),
+    });
+
+    queryClient.prefetchQuery({
+      queryKey: ["data", page + 1, limit],
+      queryFn: () => fetchTasks(page + 1, limit),
```

Are we doing verification at some point that this (`page + 1`) page exists? Or is it handled automatically by `prefetchQuery`?

---

Thank you for catching this. the prefetch will fail and it would be silently ignored. Even if this is not a big issue I decided to check if the page exists before prefetching